### PR TITLE
Allow {:system, varname} format for configuration

### PR DESCRIPTION
**Pull request by stavro**
Would love to get Exq supporting the `{:system, "VARNAME"}` format for runtime config...

Thoughts?

-- EDIT --

If you don't like the `{:system, ""}` approach, there are others:

* Exq can specify a fallback environment variable for URL (not from config)
* Exq can provide an [on_init callback similar to Phoenix](https://github.com/phoenixframework/phoenix/blob/7fbf4fc0567a3276c978b434c4f0c658228af276/lib/phoenix/endpoint.ex#L63-L80) to update config when the application starts.
* Exq can [rely less on compile-time config and more on arguments](https://elixirforum.com/t/mix-config-evolutions/4423/19)

If you think you'd merge this, I can add additional tests for when the environment variable is an invalid value (which is why coveralls says the code coverage went down)

### CODE REVIEW

**Comment by stavro** [comment]
@akira If you think you'd merge this, I can add additional tests for when the environment variable is an invalid value (which is why coveralls says the code coverage went down)

**Comment by akira** [comment]
@stavro We can support `{:system, "VARNAME"}` format. 

I haven't looked at PR in detail yet, just curious, there is also hex package: https://github.com/Nebo15/confex/, does that simplify implementation?

**Comment by stavro** [comment]
@akira I'm not sure that the logic is complex enough to warrant the addition of a dependency, but if you feel the additional configuration options granted by `confex` are worth it, I can certainly add it in.

File: lib/exq/support/opts.ex [hunk 1]
```diff
@@ -42,6 +42,21 @@ defmodule Exq.Support.Opts do
     "#{name}.Redis.Client" |> String.to_atom
   end
 
+  defp coerce_integer(value) when is_integer(value) do
```
Comment (akira) [inline lib/exq/support/opts.ex]: Would it be better to move this somewhere in utils?  Maybe config?

**Comment by akira** [comment]
@stavro looking at it, I think this PR as is will work. Maybe can add the additional test with invalid env variable. 

Thanks!

**Comment by stavro** [comment]
PR updated - though coveralls is having some issues (I force pushed an update to try and get it to wake up again, but no luck).

**Comment by akira** [comment]
Looks great, thanks for the PR Sean!!  👍  👏

**Pull request merged**